src/SmartTransactionsController.ts

```diff
@@ -576,7 +570,7 @@ export default class SmartTransactionsController extends BaseController<
       this.updateSmartTransaction({
         chainId,
         nonceDetails,
-        metamaskNetworkId,
+        metamaskNetworkId: chainId.replace(/^0x/u, ''),
```

Comment:
Oh, whoops, this isn't right. We need to convert the chain id from hex to decimal. I'll fix this.

Reply:
Sorry I missed this line. What is this `metamaskNetworkId` used for - do we need it at all?

Reply:
I'm not 100% sure, but I am assuming that the transaction objects that this controller stores include `metamaskNetworkId` because that property is a part of the TransactionMeta interface in the extension. It's used all over the place there, but in particular the `tx-state-manager` also includes/handles this property, e.g.: https://github.com/MetaMask/metamask-extension/blob/90ae4980b3aca3f2610a24ade06cf7e6db928209/app/scripts/controllers/transactions/tx-state-manager.js#L131 and also https://github.com/MetaMask/metamask-extension/blob/90ae4980b3aca3f2610a24ade06cf7e6db928209/app/scripts/controllers/transactions/tx-state-manager.js#L278. As you can see in the second link, I don't think it's a required property, but I tried to get rid of it as a part of https://github.com/MetaMask/metamask-extension/pull/17556 and wasn't sure whether it would break anything. I would assume the same applies to this transaction. But I'm happy to investigate this.

Reply:
Hmm. It might be safer to try to get rid of it, rather than convert `chainId` to `networkId` like this. They aren't equivalent, so this would lead to transactions on some networks having the wrong `networkId` assigned.

At least it would in theory - I don't think the smart transaction controller is used on any such networks.

Reply:
Okay, I've looked through the extension, mobile, and core codebases, and the only places where `metamaskNetworkId` seems to be used are:

* Transaction activity logging: https://github.com/MetaMask/metamask-extension/blob/987daee85483c38ec9592a05fb8efe1f0e71eda7/ui/components/app/transaction-activity-log/transaction-activity-log.util.js#L101
* Transaction metrics: https://github.com/MetaMask/metamask-extension/blob/987daee85483c38ec9592a05fb8efe1f0e71eda7/app/scripts/controllers/transactions/index.js#L2305
* Removal of old transactions when adding a transaction: https://github.com/MetaMask/metamask-extension/blob/987daee85483c38ec9592a05fb8efe1f0e71eda7/app/scripts/controllers/transactions/tx-state-manager.js#L278
* Optimism: https://github.com/MetaMask/metamask-extension/blob/987daee85483c38ec9592a05fb8efe1f0e71eda7/ui/helpers/utils/optimism/buildUnserializedTransaction.js#L22 (and similar in mobile too)

Except for Optimism it seems like these usages are either insignificant or can be replaced with `chainId` in most cases. Of course there are exceptions to this rule, so it depends on the network. In this case, if you feel confident that smart transactions won't be used on a network that doesn't have a proper chain ID then I can remove this.

But just to make sure, @dan437, do you have any thoughts on this?